shared_model/validators/transaction_validator.hpp

```diff
@@ -265,7 +266,22 @@ namespace shared_model {
           answer.addReason(std::move(tx_reason));
         }
 
+        constexpr auto cmd_begin = iroha::protocol::Command::kAddAssetQuantity;
+        constexpr auto cmd_end = cmd_begin
+            + boost::mpl::size<shared_model::interface::Command::
+                                   CommandListType>::type::value;
         for (const auto &command : tx.commands()) {
+          auto cmd_case =
+              static_cast<const shared_model::proto::Command &>(command)
+                  .getTransport()
+                  .command_case();
+          if (cmd_begin > cmd_case || cmd_case >= cmd_end) {
```

Comment:
Same as in query, `getTransport().has_command()` is simpler and covers the desired case.